### PR TITLE
Change relationship to LayeredOver and updated Service Probe

### PR DESCRIPTION
**Pull request by DongyiYang**
Fixed the supply chain error. The relationship between `VirtualApplication` and `Application` should be ***[LayeredOver](https://github.com/turbonomic/kubeturbo/compare/master...DongyiYang:SvcProbeChange?expand=1#diff-36ab37f1cb5a888be5e9eeeaadb59306R152)*** .
Fixed the bug that service always buys from the same provider by updating SDK dependency.
Updated ServiceProbe variables' names and typos.

### CODE REVIEW

File: pkg/discovery/probe/service_probe.go [hunk 1]
```diff
@@ -115,33 +117,34 @@ func (svcProbe *ServiceProbe) ParseService(serviceList []*api.Service, endpointL
 	}
 
 	for _, service := range serviceList {
+		serviceClusterID := service.Namespace + "/" + service.Name
 		podClusterIDs := svcProbe.findPodEndpoints(service, endpointMap)
 		if len(podClusterIDs) < 1 {
+			glog.V(3).Infof("%s is a standalone serive without any enpoint pod.", serviceClusterID)
```
Comment (chlam4) [inline pkg/discovery/probe/service_probe.go]: A typo - 'serive'

File: pkg/discovery/probe/service_probe.go [hunk 2]
```diff
@@ -115,33 +117,34 @@ func (svcProbe *ServiceProbe) ParseService(serviceList []*api.Service, endpointL
...
 		}
-		glog.V(4).Infof("service %s has the following pod as endpoints %v",
-			service.Namespace+"/"+service.Name, podClusterIDs)
+		glog.V(4).Infof("service %s has the following pod as endpoints %v", serviceClusterID, podClusterIDs)
 
 		// create the commodities bought map.
 		commoditiesBoughtMap, err := svcProbe.getCommoditiesBought(podClusterIDs)
 		if err != nil {
-			return nil, err
+			glog.Errorf("Failed to build commoditeis bought by service %s: %s", serviceClusterID, err)
```
Comment (chlam4) [inline pkg/discovery/probe/service_probe.go]: Another typo - 'commoditeis'.

File: pkg/discovery/probe/service_probe.go (hunk 1 above)
Comment (DongyiYang) [inline pkg/discovery/probe/service_probe.go]: fixed

File: pkg/discovery/probe/service_probe.go (hunk 2 above)
Comment (DongyiYang) [inline pkg/discovery/probe/service_probe.go]: fixed

**Pull request merged**